src/mpid/ch4/shm/ipc/xpmem/xpmem_pre.h

```diff
@@ -8,17 +8,20 @@
 
 #ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
 #include <xpmem.h>
+#include "xpmem_types.h"
 #endif
 
-typedef struct MPIDI_XPMEMI_seg MPIDI_XPMEMI_seg_t;
-
 /* memory handle definition */
 typedef struct {
     uint64_t src_offset;
 } MPIDI_XPMEM_mem_handle_t;
 
 typedef struct {
+#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
     MPIDI_XPMEMI_seg_t *seg_ptr;
+#else
+    char dummy;                 /* seg_ptr is for xpmem internal use */
+#endif
```

Comment:
Seems weird to expose an internal structure this way.  I'd recommend using a `void *` here, and internally casting it to `MPIDI_XPMEMI_seg_t *` internally.

Reply:
Fixed. Can you check again?